quizzes/views.py

```diff
@@ -40,17 +43,97 @@ class QuestionDetailView(generic.DetailView):
 
 def feedback(request, quiz_id):
     quiz = get_object_or_404(Quiz, pk=quiz_id)
+    feed_dict = request.session[quiz.session_feedback()]
     norm_scores_dict = request.session[quiz.session_norm_data()]
     number_of_sections = 2
     sorted_scores = {k: v for k, v in sorted(norm_scores_dict.items(), key=lambda item: item[1])[:number_of_sections]}
     return render(request, 'quizzes/feedback.html', {
         'quiz': quiz,
-        'feed_dict': request.session[quiz.session_feedback()],
-        'norm_scores': request.session[quiz.session_norm_data()],
+        'feed_dict': feed_dict,
+        'norm_scores': norm_scores_dict,
         'sorted_scores': sorted_scores,
         })
 
 
+@permission_required('admin.can_add_log_entry')
+def quiz_upload(request):
+    """
+    This is a function that lets the superuser upload a CSV to create a new quiz.
+    """
+    template = "quizzes/quiz_upload.html"
+    prompt = {
+        'headerorder': 'Order of the CSV header should be Quiz name, pub_date, description',
+        'order': 'Order for rest of CSV should be Quiz name, '
+                 'category_name, question_text, answer_text, answer_weight, feedback_text'
+    }
+    if request.method == 'GET':
+        return render(request, template, prompt)
+
+    csv_file = request.FILES['file']
+
+    if not csv_file.name.endswith('.csv'):
+        messages.error(request,'This is not a csv file')
+
+    quiz_contents = csv_file.read().decode('UTF-8')
+    io_string = io.StringIO(quiz_contents)
+    for row in csv.reader(io_string, delimiter= ',',quotechar="|"):
+        if len(row) <= 3:
```

Comment:
It seems like the code would error out if a row in the CSV file if len(row) == 4 or 5. I'm not sure what the format of the CSV file is, but be sure to add error catching if this is a possibility (if there is no way for this to occur, just ignore this comment).

Reply:
Now that I reread the function docstring, I understand the CSV format now. If it is possible for a differently formatted CSV file to be used for this, be sure to add error catching still.

Reply:
Ahh good point. I'll look into if there is another way handle the first row in the CSV too.